specifyweb/frontend/js_src/lib/components/QueryBuilder/Formatter.tsx

```diff
@@ -23,42 +25,65 @@ export function QueryFieldFormatter({
     () =>
       // Some code duplication, but required by TypeScript
       (type === 'formatter'
-        ? formatters?.formatters.map(({ table, name, title }) => ({
+        ? formatters?.formatters.map(({ table, name, title, isDefault }) => ({
             table,
             name,
             title,
+            isDefault,
           }))
-        : formatters?.aggregators.map(({ table, name, title }) => ({
+        : formatters?.aggregators.map(({ table, name, title, isDefault }) => ({
             table,
             name,
             title,
+            isDefault,
           }))
       )
         ?.filter(({ table }) => table?.name === tableName)
-        .map(({ name, title = name }) => ({ name, title })),
+        .map(({ name, title = name, isDefault }) => ({
+          name,
+          title,
+          isDefault,
+        })),
     [type, formatters, tableName]
   );
+
+  const [formatterSelectIsOpen, toggleFormatterSelect] = React.useState(false);
+
   return availableFormatters === undefined ? (
     typeof formatter === 'string' ? (
       <>{commonText.loading()}</>
     ) : null
   ) : availableFormatters.length > 1 ? (
     <>
-      {mappingElementDivider}
-      <div>
-        <Select
-          disabled={handleChange === undefined}
-          value={formatter}
-          onValueChange={handleChange}
-        >
-          <option />
-          {availableFormatters.map(({ name, title }, index) => (
-            <option key={index} value={name}>
-              {title}
-            </option>
-          ))}
-        </Select>
-      </div>
+      <Button.Small
+        aria-label={queryText.chooseFormatter()}
+        className={`${
+          availableFormatters.find((selected) => selected.name === formatter)
+            ?.isDefault || formatter === undefined
+            ? 'bg-white dark:bg-neutral-600'
+            : 'bg-yellow-250 dark:bg-yellow-900 '
+        }`}
+        title={queryText.chooseFormatter()}
+        onClick={() => toggleFormatterSelect(!formatterSelectIsOpen)}
+      >
+        {icons.cog}
+      </Button.Small>
+      {formatterSelectIsOpen && (
```

Comment:
(optional)
this might be more user-friendly if implemented using https://headlessui.com/react/menu or https://headlessui.com/react/popover

Reply:
This is what I was hoping for with my initial description